internal/resource/config/serversettings/server_settings_system_keys_resource.go

```diff
@@ -0,0 +1,396 @@
+						Computed:    true,
+						Optional:    false,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+			},
+			"pending": schema.SingleNestedAttribute{
+				Description: "Pending SystemKeys Secrets that are used in cryptographic operations to generate and consume internal tokens.",
+				Required:    true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"creation_date": schema.StringAttribute{
+						Description: "Creation time of the key.",
+						Computed:    true,
+						Optional:    false,
+						Required:    false,
+					},
+					"encrypted_key_data": schema.StringAttribute{
+						Description: "The system key encrypted.",
+						Computed:    true,
+						Optional:    true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+					"key_data": schema.StringAttribute{
+						Description: "The clear text system key base 64 encoded. The system key must be 32 bytes before base 64 encoding",
+						Computed:    true,
+						Optional:    false,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	resp.Schema = schema
+}
+
+func addServerSettingsSystemKeysFields(ctx context.Context, addRequest *client.SystemKeys, plan serverSettingsSystemKeysResourceModel) error {
+
+	if internaltypes.IsDefined(plan.Current) {
+		currentAttrs := plan.Current.Attributes()
+		encryptedKeyDataAttrcurrent := currentAttrs["encrypted_key_data"]
+		if internaltypes.IsNonEmptyString(encryptedKeyDataAttrcurrent.(types.String)) {
+			addRequest.Current = *client.NewSystemKey()
+			currentEncryptedKeyData := internaltypes.ConvertToPrimitive(encryptedKeyDataAttrcurrent).(string)
+			addRequest.Current.EncryptedKeyData = &currentEncryptedKeyData
+		}
+	}
+	if internaltypes.IsDefined(plan.Previous) {
+		previousAttrs := plan.Previous.Attributes()
+		encryptedKeyDataAttrPrevious := previousAttrs["encrypted_key_data"]
+		if internaltypes.IsNonEmptyString(encryptedKeyDataAttrPrevious.(types.String)) {
+			addRequest.Previous = client.NewSystemKey()
+			previousEncryptedKeyData := internaltypes.ConvertToPrimitive(encryptedKeyDataAttrPrevious).(string)
+			addRequest.Previous.EncryptedKeyData = &previousEncryptedKeyData
+		}
+	}
+	if internaltypes.IsDefined(plan.Pending) {
+		pendingAttrs := plan.Pending.Attributes()
+		encryptedKeyDataAttrPending := pendingAttrs["encrypted_key_data"]
+		if internaltypes.IsNonEmptyString(encryptedKeyDataAttrPending.(types.String)) {
+			addRequest.Pending = *client.NewSystemKey()
+			pendingEncryptedKeyData := internaltypes.ConvertToPrimitive(encryptedKeyDataAttrPending).(string)
+			addRequest.Pending.EncryptedKeyData = &pendingEncryptedKeyData
+		}
+	}
+	return nil
```

Comment:
If the method always returns nil then there's no reason to have the method return anything at all